### ISSUE TITLE
Use fs.promises where possible

### ISSUE THREAD

**Issue by benjamingr** [normalize/mz#64]
Node now ships with an fs promise API, it would be cool if mz would detect if that API is available (optionally?) and use it if it's there.

**Comment by RyanZim** [comment]
As of Node v10.1, usage of `fs.promises` raises an `ExperementalWarning` in the console, I don't think we should use that until it's stable.

**Comment by benjamingr** [comment]
Fair enough - feel free to close or keep open as a tracking issue - we intend to make it non-experimental as soon as we've gathered enough user feedback about it :)

Thanks for the quick response.

**Comment by RyanZim** [comment]
Not on the team, just commenting here.

**Comment by LinusU** [comment]
What @RyanZim said 😄 

Yeah, we really don't want to trigger `ExperementalWarning` for our users, so let's wait until it's stable ☺️

**Comment by RyanZim** [comment]
@LinusU Speaking of which, https://github.com/normalize/mz/issues/63 should be fixed :wink:

**Comment by ExE-Boss** [comment]
Usage&nbsp;of&nbsp;`fs.promises` would&nbsp;be&nbsp;a&nbsp;breaking&nbsp;change, since&nbsp;`fs.promises` returns&nbsp;an&nbsp;object for&nbsp;callbacks&nbsp;returning more&nbsp;than&nbsp;one&nbsp;item, whereas&nbsp;`mz` returns&nbsp;an&nbsp;array in&nbsp;those&nbsp;cases.

Compare&nbsp;the&nbsp;return&nbsp;type of&nbsp;`fs.promises.write(…)` to&nbsp;that&nbsp;of&nbsp;`mz/fs.write(…)`.
